src/main/java/uk/gov/pay/api/filter/ratelimit/LocalRateLimiter.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.pay.api.filter.ratelimit;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class LocalRateLimiter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocalRateLimiter.class);
+
+    private final int perMillis;
+    private final Cache<String, RateLimit> cache;
+
+    public LocalRateLimiter(int perMillis) {
+        this.perMillis = perMillis;
+        this.cache = CacheBuilder.newBuilder()
+                .expireAfterAccess(perMillis, TimeUnit.MILLISECONDS)
+                .build();
+    }
+
+    void checkRateOf(String key, int rate) throws RateLimitException {
+        try {
+            cache.get(key, () -> new RateLimit(rate, perMillis)).updateAllowance();
+        } catch (ExecutionException e) {
+            //ExecutionException is thrown when the valueLoader throws a checked exception.
+            //We just create a new instance so no exceptions will be thrown, this should never happen.
```

Comment:
I don't understand this comment. what's `valueLoader`? and how are we creating a new instance? I suggest some re-wording here.

Reply:
Updated comment for value loader and new instance